### PR TITLE
Restructure commands hierarchy

### PR DESCRIPTION
**Pull request by tjololo**
## Description
Restructure the commands hierarchy and try to make the code match the same hierarchy

Hierarchy now:
```shell
➜  ~ altinn-studio -h
Description:
  Command line interface for working with Altinn 3 Applications

Usage:
  altinn-studio [command] [options]

Options:
  --version       Show version information
  -?, -h, --help  Show help and usage information

Commands:
  upgrade  Upgrade an app
  version  Print version of altinn-studio cli
```
```shell
➜  ~ altinn-studio upgrade -h
Description:
  Upgrade an app

Usage:
  altinn-studio upgrade [command] [options]

Options:
  --folder <folder>  The project folder to read [default: CurrentDirectory]
  -?, -h, --help     Show help and usage information

Commands:
  frontend  Upgrade an app from using App-Frontend v3 to v4
  backend   Upgrade an app from app-lib-dotnet v7 to v8
```
```shell
➜  ~ altinn-studio upgrade backend -h
Description:
  Upgrade an app from app-lib-dotnet v7 to v8

Usage:
  altinn-studio upgrade backend [options]

Options:
  --folder <folder>                          The project folder to read [default: CurrentDirectory]
  --project <project>                        The project file to read relative to --folder [default: App/App.csproj]
  --process <process>                        The process file to read relative to --folder [default: App/config/process/process.bpmn]
  --appsettings-folder <appsettings-folder>  The folder where the appsettings.*.json files are located [default: App]
  --target-version <target-version>          The target version to upgrade to [default: 8.0.0-preview.11]
  --target-framework <target-framework>      The target dotnet framework version to upgrade to [default: net8.0]
  --skip-csproj-upgrade                      Skip csproj file upgrade [default: False]
  --skip-dockerfile-upgrade                  Skip Dockerfile upgrade [default: False]
  --skip-code-upgrade                        Skip code upgrade [default: False]
  --skip-process-upgrade                     Skip process file upgrade [default: False]
  --skip-appsettings-upgrade                 Skip appsettings file upgrade [default: False]
  -?, -h, --help                             Show help and usage information

```
```shell
➜  ~ altinn-studio upgrade frontend -h
Description:
  Upgrade an app from using App-Frontend v3 to v4

Usage:
  altinn-studio upgrade frontend [options]

Options:
  --folder <folder>                              The project folder to read [default: CurrentDirectory]
  --target-version <target-version>              The target version to upgrade to [default: 4.0.0-rc1]
  --index-file <index-file>                      The name of the Index.cshtml file relative to --folder [default: App/views/Home/Index.cshtml]
  --skip-index-file-upgrade                      Skip Index.cshtml upgrade [default: False]
  --ui-folder <ui-folder>                        The folder containing layout files relative to --folder [default: App/ui/]
  --texts-folder <texts-folder>                  The folder containing text files relative to --folder [default: App/config/texts/]
  --layout-set-name <layout-set-name>            The name of the layout set to be created [default: form]
  --application-metadata <application-metadata>  The path of the applicationmetadata.json file relative to --folder [default: 
                                                 App/config/applicationmetadata.json]
  --skip-layout-set-upgrade                      Skip layout set upgrade [default: False]
  --skip-settings-upgrade                        Skip layout settings upgrade [default: False]
  --skip-layout-upgrade                          Skip layout files upgrade [default: False]
  --preserve-default-triggers                    Preserve default schema and component validation triggers [default: False]
  --convert-group-titles                         Convert 'title' in repeating groups to 'summaryTitle' [default: False]
  --skip-schema-ref-upgrade                      Skip schema reference upgrade [default: False]
  --skip-footer-upgrade                          Skip footer upgrade [default: False]
  --skip-checks                                  Skip checks [default: False]
  -?, -h, --help                                 Show help and usage information



```

Cleanup some warnings at the same time

## Related Issue(s)
- #{issue number}

## Verification
- [ ] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [ ] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/altinn-studio-cli/Makefile [hunk 1]
```diff
@@ -16,5 +16,8 @@ clean:
 install-locally: pack
 	dotnet tool install --global --add-source ./publish Altinn.Studio.Cli --prerelease
 	
+reinstall-locally: uninstall-locally clean pack
+	dotnet tool install --global --add-source ./publish Altinn.Studio.Cli --prerelease
+
```
Comment (bjosttveit) [inline src/altinn-studio-cli/Makefile:21]: 🤩

Comment (tjololo) [reply src/altinn-studio-cli/Makefile:21]: felt a need ;)

**Comment by bjosttveit** [comment]
Have you considered the return code for `BackendUpgrade`? I am not sure what the best practice is in this area but to make it work in `FrontendUpgrade` which was outside of `Program.cs` I added an `Environment.Exit(returnCode);` at the end of the command handler.

**Comment by tjololo** [comment]
> Have you considered the return code for `BackendUpgrade`? I am not sure what the best practice is in this area but to make it work in `FrontendUpgrade` which was outside of `Program.cs` I added an `Environment.Exit(returnCode);` at the end of the command handler.

Forgot about the returncode. Have used the Environment.Exit(returnCode); method as you have in `upgrade frontend`

**Review by bjosttveit — APPROVED** [review]
Nice and clean 😎

**Pull request merged**